aws-cpp-sdk-core/include/aws/core/internal/AWSHttpResourceClient.h

```diff
@@ -29,6 +30,15 @@ namespace Aws
 
     namespace Internal
     {
+        /**
+         * This mutable structure is used to configure AWSHttpResourceClient.
+         * Default values can only be overwritten prior to passing to the client constructors.
+         */
+        struct HttpResourceClientConfiguration: public Client::ClientConfiguration
+        {
+            HttpResourceClientConfiguration();
+        };
+
```

Comment:
This is the only wrinkle I see in this PR.

I'd rather not introduce another type just for convenience.
I prefer having two constructors one that takes `ClientConfiguration` parameter and another that does not. Using the constructor without the client configuration would set the defaults you have in this PR.

Thoughts?

Reply:
Sounds reasonable. I’ll update my PR with it.